lib/src/ast/selector/pseudo.dart

```diff
@@ -80,19 +79,30 @@ class PseudoSelector extends SimpleSelector {
   /// both non-`null`, the selector follows the argument.
   final SelectorList? selector;
 
-  int get minSpecificity {
-    if (_minSpecificity == null) _computeSpecificity();
-    return _minSpecificity!;
-  }
-
-  int? _minSpecificity;
-
-  int get maxSpecificity {
-    if (_maxSpecificity == null) _computeSpecificity();
-    return _maxSpecificity!;
-  }
-
-  int? _maxSpecificity;
+  late final int specificity = () {
+    if (isElement) return 1;
+    var selector = this.selector;
+    if (selector == null) return super.specificity;
+
+    // https://drafts.csswg.org/selectors/#specificity-rules
```

Comment:
Linking to https://www.w3.org/TR/selectors-4/#specificity-rules seems better than using the draft IMO.